homeassistant/components/hdmi_cec.py

```diff
@@ -1,121 +1,290 @@
+                    hdmi_network.send_command(
+                        KeyReleaseCommand(dst=ADDR_AUDIOSYSTEM))
+                _LOGGER.info("Volume increased %d times", att)
+            elif cmd == CMD_DOWN:
+                for _ in range(att):
+                    hdmi_network.send_command(
+                        KeyPressCommand(KEY_VOLUME_DOWN, dst=ADDR_AUDIOSYSTEM))
+                    hdmi_network.send_command(
+                        KeyReleaseCommand(dst=ADDR_AUDIOSYSTEM))
+                _LOGGER.info("Volume deceased %d times", att)
+            elif cmd == CMD_MUTE:
+                hdmi_network.send_command(
+                    KeyPressCommand(KEY_MUTE, dst=ADDR_AUDIOSYSTEM))
+                hdmi_network.send_command(
+                    KeyReleaseCommand(dst=ADDR_AUDIOSYSTEM))
+                _LOGGER.info("Audio muted")
+            else:
+                _LOGGER.warning("Unknown command %s", cmd)
 
-    def _start_cec(event):
-        """Open CEC adapter."""
-        adapters = _CEC.DetectAdapters()
-        if len(adapters) == 0:
-            _LOGGER.error("No CEC adapter found")
-            return
-
-        if _CEC.Open(adapters[0].strComName):
-            hass.services.register(DOMAIN, SERVICE_POWER_ON, _power_on)
-            hass.services.register(DOMAIN, SERVICE_STANDBY, _standby)
-            hass.services.register(DOMAIN, SERVICE_SELECT_DEVICE,
-                                   _select_device)
+    @callback
+    def _tx(call):
+        """Send CEC command."""
+        data = call.data
+        if ATTR_RAW in data:
+            command = CecCommand(data[ATTR_RAW])
         else:
-            _LOGGER.error("Failed to open adapter")
+            if ATTR_SRC in data:
+                src = data[ATTR_SRC]
+            else:
+                src = ADDR_UNREGISTERED
+            if ATTR_DST in data:
+                dst = data[ATTR_DST]
+            else:
+                dst = ADDR_BROADCAST
+            if ATTR_CMD in data:
+                cmd = data[ATTR_CMD]
+            else:
+                _LOGGER.error("Attribute 'cmd' is missing")
+                return False
+            if ATTR_ATT in data:
+                if isinstance(data[ATTR_ATT], (list,)):
+                    att = data[ATTR_ATT]
+                else:
+                    att = reduce(lambda x, y: "%s:%x" % (x, y), data[ATTR_ATT])
+            else:
+                att = ""
+            command = CecCommand(cmd, dst, src, att)
+        hdmi_network.send_command(command)
+
+    @callback
+    def _update(call):
+        """
+        Callback called when device update is needed.
+
+        - called by service, requests CEC network to update data.
+        """
+        hdmi_network.scan()
+
+    @callback
+    def _new_device(device):
+        """Called when new device is detected by HDMI network."""
+        discovery.load_platform(hass, "switch", DOMAIN,
+                                discovered={ATTR_NEW: [device]},
```

Comment:
So should I put device into hasss.data with my unique key and then retrieve it by this key which will be send over load_platform?